functions/slack/app/format.go

```diff
@@ -0,0 +1,74 @@
+// Copyright 2018, Google, LLC.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// [START functions_slack_format]
+
+package slack
+
+import (
+	"fmt"
+
+	"google.golang.org/api/kgsearch/v1"
+)
+
+func formatSlackMessage(query string, response *kgsearch.SearchResponse) (*Message, error) {
+	var entity interface{}
+	if len(response.ItemListElement) > 0 {
+		entity = response.ItemListElement[0]
+	}
+	resp, ok := entity.(map[string]interface{})
+	if !ok {
+		return nil, fmt.Errorf("error formatting response entity")
+	}
+	result, ok := resp["result"].(map[string]interface{})
+	if !ok {
+		return nil, fmt.Errorf("error formatting response result")
+	}
+	message := &Message{
+		ResponseType: "in_channel",
+		Text:         fmt.Sprintf("Query: %s", query),
+		Attachments:  []attachment{},
+	}
+
+	attachment := attachment{}
```

Comment:
This pre-declaration is also a little odd. Can append and return directly, rather than doing this indirectly. Also, should use `len` to check if there is a result after making the change above (which will return an empty map).

```go
if len(entity) == 0 {
	message.Attachments = {attachment{Text: "No results match your query."}}
	return message, nil
}
if name, ok := ...
```